From 12d1057fba45b534d0cfbbcc900e5032e314b150 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 11 Apr 2017 08:29:47 +0100
Subject: [PATCH] ocaml: Add Bytes.sub_string function to Bytes compat module.

This is only used when OCaml < 4.02 (eg. RHEL 7).  For more recent
versions the real Bytes.sub_string function is used instead.

(cherry picked from commit bb846887de7a252204152c3e6df80d39a5ea33af)
---
 m4/guestfs_ocaml.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/m4/guestfs_ocaml.m4 b/m4/guestfs_ocaml.m4
index 0479e70ab..d8bd89f00 100644
--- a/m4/guestfs_ocaml.m4
+++ b/m4/guestfs_ocaml.m4
@@ -120,6 +120,7 @@ AS_IF([test "x$have_Bytes_module" = "xno"],[
 include String
 let of_string = String.copy
 let to_string = String.copy
+let sub_string = String.sub
 EOF
     ln -s ../generator/bytes.ml mllib/bytes.ml
     OCAML_GENERATOR_BYTES_COMPAT_CMO='$(top_builddir)/generator/bytes.cmo'
-- 
2.13.4

